docs/core/docker/publish-as-container.md

```diff
@@ -0,0 +1,349 @@
+
+ContainerLabel items have two properties:
+
+* Include
+  * The key of the label
+* Value
+  * The value of the label - this may be empty
+
+See [default container labels](#default-container-labels) for a list of labels that are created by default.
+
+```xml
+<ItemGroup>
+    <ContainerLabel Include="org.contoso.businessunit" Value="contoso-university" />
+<ItemGroup>
+```
+
+## `ContainerEnvironmentVariable`
+
+This item adds a new environment variable to the container. Environment variables will be accessible to the application running in the container immediately, and are often used to change the runtime behavior of the running application.
+
+ContainerEnvironmentVariable items have two properties:
+
+* Include
+  * The name of the environment variable
+* Value
+  * The value of the environment variable
+
+```xml
+<ItemGroup>
+  <ContainerEnvironmentVariable Include="LOGGER_VERBOSITY" Value="Trace" />
+</ItemGroup>
+```
+
+## `ContainerEntrypoint`
+
+This item can be used to customize the entrypoint of the container - the binary that is run by default when the container is started.
+
+By default, for builds that create an executable binary that binary is set as the ContainerEntrypoint. For builds that do not create an executable binary `dotnet path/to/application.dll` is used as the ContainerEntrypoint.
+
+ContainerEntrypoint items have one property:
+
+* Include
+  * The command, option, or argument to use in the entrypoint command
+
+```xml
+<ItemGroup Label="Entrypoint Assignment">
+  <!-- This is how you would start the dotnet ef tool in your container -->
+  <ContainerEntrypoint Include="dotnet" />
+  <ContainerEntrypoint Include="ef" />
+
+  <!-- This shorthand syntax means the same thing - note the semicolon separating the tokens. -->
+  <ContainerEntrypoint Include="dotnet;ef" />
+</ItemGroup>
+```
+
+## `ContainerEntrypointArgs`
+
+This item controls the default arguments provided to the `ContainerEntrypoint`. This should be used when the ContainerEntrypoint is a program that the user might want to use on its own.
+
+By default, no ContainerEntrypointArgs are created on your behalf.
+
+ContainerEntrypointArg items have one property:
+
+* Include
+  * The option or argument to apply to the ContainerEntrypoint command
+
+```xml
+<ItemGroup>
+  <!-- Assuming the ContainerEntrypoint defined above, this would be the way to update the database by default, but let the user run a different EF command. -->
+  <ContainerEntrypointArgs Include="database" />
+  <ContainerEntrypointArgs Include="update" />
+
+  <!-- This is the shorthand syntax for the same idea -->
+  <ContainerEntrypointArgs Include="database;update" />
+</ItemGroup>
+```
+
+### Default container labels
+
+Labels are often used to provide consistent metadata on container images. This package provides some default labels to encourage better maintainability of the generated images.
```

Comment:
We want to add a bunch more, should we link to the issue for that and ask for feedback on labels that users want? The matching issue is https://github.com/dotnet/sdk-container-builds/issues/96

Reply:
Add link